File: src/main/java/uk/co/threebugs/mochiwhattotrade3/account/AccountMapper.java
package uk.co.threebugs.mochiwhattotrade3.account;

import org.mapstruct.Mapper;

import java.nio.file.Path;
import java.nio.file.Paths;

@Mapper(componentModel = "spring")
public interface AccountMapper {

    AccountDto toDto(Account account);

    Account toEntity(AccountDto accountDto);

    default String pathToString(Path path) {
        return path == null ? null : path.toString();
    }

    default Path stringToPath(String path) {
        return path == null ? null : Paths.get(path);
    }

}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/account/AccountMapper.kt
package uk.co.threebugs.mochiwhattotrade3.account

import org.mapstruct.Mapper
import java.nio.file.Path
import java.nio.file.Paths

@Mapper(componentModel = "spring")
abstract class AccountMapper {

    abstract fun toDto(account: Account): AccountDto
    abstract fun toEntity(accountDto: AccountDto): Account

    fun toPath(path: String?): Path? {
        return path?.let { Path.of(it) }
    }

    fun toString(path: Path?): String? {
        return path?.toString()
    }
}
